Rename two internal variables for readability

zmwangx opened the pull request:
This PR is highly subjective, but anyway, I found two confusingly named (to me) variables that caused me a bit of trouble when reading the code — I had to go back and forth between several functions to understand what they are doing. Here they are:

* `prompt_pure_git_delay_dirty_check` sounds like a boolean, but holds the (finish) timestamp of the last dirty check (if the dirty check took more than 2 seconds).

  I renamed it to `prompt_pure_git_last_dirty_check_timestamp`.

* Local variable `dirty_check` holds the elapsed time in seconds since the last dirty check (if the timestamp is set), but the variable name is too generic to make sense of.

  I renamed it to `time_since_last_dirty_check`.

mafredri commented on pure.zsh:
```diff
@@ -227,9 +227,9 @@ prompt_pure_async_callback() {
-			# prompt is rendered before the variable is (potentially) set
-			(( $exec_time > 2 )) && prompt_pure_git_delay_dirty_check=$EPOCHSECONDS
+			# when prompt_pure_git_last_dirty_check_timestamp is set, the git info is displayed in a different color,
+			# this is why the prompt is rendered before the variable is (potentially) set
```
Hmm, your editing this comment made me realize that it itself is very confusing. I think it would be best to remove the whole `this is why the prompt is rendered before the variable is (potentially) set` since prompt rendering is not a result of this variable being set or not.

zmwangx replied:
@mafredri Hmm, but I think it is? `git_color` will be `red` if this variable is set (see https://github.com/zmwangx/pure/blob/5b522d86dad019b08e2684114fc9716266d2ffc6/pure.zsh#L85-L86).

mafredri replied:
No, not in the sense that the comment is saying. I think something along these lines is more appropriate:
```sh
# When prompt_pure_git_delay_dirty_check_timestamp is set, the git info is displayed in a different color.
# To distinguish between a "fresh" and a "cached" result, the preprompt is rendered before setting this
# variable. Thus, only upon next rendering of the preprompt will the result appear in a different color.
```

mafredri commented:
Good catch with the variable names, I agree that the renamed ones make a lot more sense, thanks! Would you mind still modifying the one comment I commented on?

zmwangx commented:
I thought the original comment was already pretty understandable (I read the comment, went back to render, and immediately understood what was going on), but sure, why not. Changed and rebased. Any other problems? I see that you were using a slightly different variable name from mine in the previous comment (`prompt_pure_git_delay_dirty_check_timestamp` as opposed to `prompt_pure_git_last_dirty_check_timestamp`).

mafredri commented:
Hmm, you are right. Now that I look at the old comment again it makes sense, momentary lapse in thought, perhaps? :). The new comment is perhaps overly verbose, but I hope it's more clear.

Regarding `prompt_pure_git_delay_dirty_check_timestamp` vs `prompt_pure_git_last_dirty_check_timestamp` I should have just copy-pasted your variable name instead of adding `_timestamp` to the old one.

Thanks for the PR!

zmwangx commented:
> Now that I look at the old comment again it makes sense, momentary lapse in thought, perhaps?

Yeah, happens all the time ;)
